articles/ros2_threat_model.md

```diff
@@ -0,0 +1,3291 @@
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="warning">2</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>14</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="warning">▲</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="warning">▲</td>
+    <td class="warning">▲</td>
+    <td>
+      <ul>
+        <li>Deployment service should be authenticated• Communication with
+the deployment service should be done over a secure channel.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr>
+    <td>An attacker intercepts the binaries sent by the depoyment service.</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td class="warning">2</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>12</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Deployment service should be authenticated• Communication with
+the deployment service should be done over a secure channel.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr>
+    <td>An attacker prevents the robot and the deployment service from
+communicating.</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td class="warning">2</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>12</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    
```

Comment:
Are you open to consider more communication channels? E.g. serial ports such as SPI or UART should be available both from the RPi. Similarly, OpenCR exposes communication ports.

Reply:
On a second reading, there're several more entry points worth mentioning here @thomas-moulard. It may be worth considering them. There're some simple aspects such as unused communication ports which can be simply disabled by configuring the embedded computer (and the underlying OS) appropriately. Would you be open for contributions that would affect (slightly to include those columns below) the threat model below? @XabierPB and @o-olalde, would you be interested on this?

Reply:
Yes, that'd be great. As with your other suggestion, it may be better to get this merged and iterate on the doc, but I don't have a very strong preference there.